### DISCUSSION TITLE
Use of PMOD

### DISCUSSION THREAD

**Discussion by K7MDL2** [cariboulabs/cariboulite#111] [General]
Has anyone looked at using PMOD pins?  Any test apps.  I found the library function to set and get, but before I write something, checking to see if it already exists somewhere by someone.  I did not see it in the supplied test apps or production utilities.

**Comment by gjg1976** [comment]
Same question as K7MDL2. 
there is no a clear API doc, but in the code there are two functions: CaribouLite::GetPmodState () and void CaribouLite::SetPmodState (uint8_t val)
then how can I set Input or Output?
the HDL defines
PMOD 
set_io io_pmod_out[0] B24 
set_io io_pmod_out[1] A31 
set_io io_pmod_out[2] B23 
set_io io_pmod_out[3] B21 
set_io io_pmod_in[0] A25 
set_io io_pmod_in[1] A26 
set_io io_pmod_in[2] A27 
set_io io_pmod_in[3] B20
and  
ioc_pmod_dir        = 5'b00011,     // PMOD connector bits IO pin direction
so it is confusing, could you clarify which PMODs are Input which ones are Output, are they I/O configurable? or fixed? 
Thank you 
Regards
